staff/marc-sarroca/e-commerce-protos/scripts/cart.js

```diff
@@ -0,0 +1,42 @@
+function Cart() {
+  this.products = [];
+}
+
+Cart.prototype.add = function(product) {
+  this.products.push(product);
+};
+
+Cart.prototype.totalPrice = function() {
+  return this.products.reduce(function(a, b) {
+    return a + b.price;
+  }, 0);
+};
+
+Cart.prototype.numberOfItems = function() {
+  return this.products.length;
+};
+
+Cart.prototype.mostExpensive = function() {
+  var items = [];
+  for (var i = 0; i < this.products.length; i++) items.push(this.products[i]);
+
+  var orderProducts = items.sort(function(a, b) {
+    return a.price - b.price;
+  });
+  return orderProducts.pop();
+};
+
+Cart.prototype.cheapest = function() {
+  var items = [];
+  for (var i = 0; i < this.products.length; i++) items.push(this.products[i]);
+
+  var orderProducts = items.sort(function(a, b) {
+    return b.price - a.price;
+  });
+  return orderProducts.pop();
+};
+Cart.prototype.numberOf = function() {
+  return this.products.reduce(function(accum, product) {
+    return product instanceof type ? ++accum : accum;
```

Comment:
where is the type variable defined? 
you need to declare as argument (in line 38) in the function value like this:

```js
Cart.prototype.numberOf = function(type) {
```

tell me if you need help with this!
come on!!

Reply:
👍